receiver/awscontainerinsightreceiver/internal/neuron/neuron_monitor_scraper_test.go

```diff
@@ -0,0 +1,393 @@
+# HELP system_memory_total_bytes System memory total_bytes bytes
+# TYPE system_memory_total_bytes gauge
+system_memory_total_bytes{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f"} 5.32523487232e+011
+# HELP system_memory_used_bytes System memory used_bytes bytes
+# TYPE system_memory_used_bytes gauge
+system_memory_used_bytes{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f"} 7.6337672192e+010
+# HELP system_swap_total_bytes System swap total_bytes bytes
+# TYPE system_swap_total_bytes gauge
+system_swap_total_bytes{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f"} 0.0
+# HELP system_swap_used_bytes System swap used_bytes bytes
+# TYPE system_swap_used_bytes gauge
+system_swap_used_bytes{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f"} 0.0
+# HELP system_vcpu_count System vCPU count
+# TYPE system_vcpu_count gauge
+system_vcpu_count{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f"} 128.0
+# HELP system_vcpu_usage_ratio System CPU utilization ratio
+# TYPE system_vcpu_usage_ratio gauge
+system_vcpu_usage_ratio{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f",usage_type="user"} 0.008199999999999999
+# HELP instance_info EC2 instance information
+# TYPE instance_info gauge
+instance_info{availability_zone="us-east-1c",instance_id="i-09db9b55e0095612f",instance_name="",instance_type="trn1n.32xlarge",region="us-east-1",subnet_id="subnet-06a7754948e8a000f"} 1.0
+`
+
+const dummyInstanceId = "i-0000000000"
+const nueronCoreID = "0"
+
+type mockHostInfoProvider struct {
+}
+
+func (m mockHostInfoProvider) GetClusterName() string {
+	return "cluster-name"
+}
+
+func (m mockHostInfoProvider) GetInstanceID() string {
+	return dummyInstanceId
+}
+
+type mockConsumer struct {
+	t             *testing.T
+	up            *bool
+	coreUtil      *bool
+	memUsed       *bool
+	instanceId    *bool
+	podName       *bool
+	containerName *bool
+	namespace     *bool
+	relabeled     *bool
+}
+
+func (m mockConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{
+		MutatesData: false,
+	}
+}
+
+func (m mockConsumer) ConsumeMetrics(_ context.Context, md pmetric.Metrics) error {
```

Comment:
In combination with some other comments I've made on test structure, I think a nicer test would be to specify exactly which metrics and labels you're looking for:

```
expected := {
    name: "neuroncore_utilization_ratio",
    labels: {
        "NodeName": "i-09db9b55e0095612f",
        ...
    },
    value: 0.1
}
```

(Or an array of those.)  Then your validation code would take the output of `GetMetrics()` and exactly match up each metric's name, labels, and value.